mqtt/mqtt-bridge/src/bridge.rs

```diff
@@ -1,31 +1,387 @@
-use tracing::info;
+use std::{collections::HashMap, marker::PhantomData, str::FromStr, time::Duration};
 
-use crate::settings::ConnectionSettings;
+use anyhow::Result;
+use async_trait::async_trait;
+use mqtt3::{proto::Publication, Event, ReceivedPublication};
+use mqtt_broker::TopicFilter;
+use tracing::{debug, info, warn};
+
+use crate::client::{EventHandler, MqttClient};
+use crate::persist::{memory::InMemoryPersist, Persist};
+use crate::settings::{ConnectionSettings, Credentials, Topic};
+
+const BATCH_SIZE: usize = 10;
 
 /// Bridge implementation that connects to local broker and remote broker and handles messages flow
 pub struct Bridge {
+    system_address: String,
+    device_id: String,
     connection_settings: ConnectionSettings,
+    forwards_map: HashMap<String, Topic>,
+    subs_map: HashMap<String, Topic>,
 }
 
 impl Bridge {
-    pub fn new(connection_settings: ConnectionSettings) -> Self {
+    pub fn new(
+        system_address: String,
+        device_id: String,
+        connection_settings: ConnectionSettings,
+    ) -> Self {
+        let forwards_map: HashMap<String, Topic> = connection_settings
+            .forwards()
+            .iter()
+            .map(move |sub| Bridge::format_key_value(sub))
+            .collect();
+
+        let subs_map: HashMap<String, Topic> = connection_settings
+            .subscriptions()
+            .iter()
+            .map(|sub| Bridge::format_key_value(sub))
+            .collect();
+
         Bridge {
```

Comment:
If you use type name inside the type definition, you can use `Self` instead of type name `Bridge` like
`Self { ... }` or `Self::foo(bar)`